staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -526,6 +530,7 @@ func (f *DeltaFIFO) Pop(process PopProcessFunc) (interface{}, error) {
 
 			f.cond.Wait()
 		}
+		isInInitialList := !f.hasSynced_locked()
```

Comment:
Things can't get demoted/preempted in the queue, so I think this isn't a problem?

A future change could get merged with a change from the initial list, if you don't want that you have to use the regular fifo. But most uses do want that, since it's more efficient.

Reply:
Why is the queue important? My read was that `f.populated` was global? I'm considering the case where:

1. `*Reflector.syncWith()` calls `*DeltaFIFO.Replace()` for all items in the original list
1. `*Reflector.ListAndWatch()` begins the watch
1. `*controller.processLoop()` begins calling `*DeltaFIFO.Pop()` to process them
1. `*Reflector.watchHandler()` calls `*DeltaFIFO.Add()` with some new object it saw in the watch stream
1. `*controller.processLoop()` finishes using `*DeltaFIFO.Pop()` to consume all objects from the initial list

The delta FIFO `HasSynced()` would return `true` after 4, even though the initial list had not been processed yet, right? Since `f.populated` gets latched once `Add()` is called. The controller relinquishes the FIFO's lock between each `Pop()`, so the `Add()` could be interleaved?

Reply:
`populated` is already set to true at the end of Replace()? It's initialPopulationCount going to zero that determines when it is synced.

Reply:
Ah, jeez, sorry - I took the comment on `HasSynced` at face value and didn't realize `Replace()` also set it. Sorry for the noise!

Reply:
no worries :)